runtime/strings/strings.cpp

```diff
@@ -0,0 +1,162 @@
+#include<assert.h>
+#include<stdlib.h>
+#include<iostream>
+
+extern "C" {
+
+// Assume assume length is passed before char *
+
+inline int get_len(const char * a) {
```

Comment:
Assumption here is that function will be passed a string with the length prefixed _before_ the pointer of the beginning of the string. This is a pretty common way to allocate things. It can be easily changed if we want.